graders/java/JUnitAutograder.java

```diff
@@ -157,6 +162,26 @@ public synchronized void dynamicTestRegistered(TestIdentifier test) {
         addProvisionalTest(test);
     }
 
+    @Override
+    public synchronized void reportingEntryPublished(TestIdentifier test, ReportEntry entry) {
+
+        if (!test.isTest())
+            return;
+
+        AutograderTest autograderTest = tests.get(test);
+        if (autograderTest == null)
+            return;
+
+        for (Map.Entry<String, String> pair : entry.getKeyValuePairs().entrySet()) {
+            // "value" is the key used by TestReporter if only a value is provided (without a key)
+            // https://junit.org/junit5/docs/5.9.1/api/org.junit.jupiter.api/org/junit/jupiter/api/TestReporter.html#publishEntry(java.lang.String)
+            if (pair.getKey().equals("value"))
+                autograderTest.output += pair.getValue() + "\n";
+            else
+                autograderTest.output += pair.getKey() + ": " + pair.getValue() + "\n";
+        }
+    }
```

Comment:
Nit (this is how it'd be in JS, maybe it's foolish to aim for consistency with that though):

```suggestion
        if (!test.isTest()) {
            return;
        }

        AutograderTest autograderTest = tests.get(test);
        if (autograderTest == null) {
            return;
        }

        for (Map.Entry<String, String> pair : entry.getKeyValuePairs().entrySet()) {
            // "value" is the key used by TestReporter if only a value is provided (without a key)
            // https://junit.org/junit5/docs/5.9.1/api/org.junit.jupiter.api/org/junit/jupiter/api/TestReporter.html#publishEntry(java.lang.String)
            if (pair.getKey().equals("value")) {
                autograderTest.output += pair.getValue() + "\n";
            } else {
                autograderTest.output += pair.getKey() + ": " + pair.getValue() + "\n";
            }
        }
    }
```

Reply:
I'm using [checkstyle](https://checkstyle.org/contributing.html#Build) for style, and this is not enforced. I'd prefer to keep it as is.

Reply:
https://dwheeler.com/essays/apple-goto-fail.html 😉 🤷

Reply:
Seems like I misconfigured checkstyle, I checked the documentation and it's supposed to enforce that too. Updates coming.

Reply:
Turns out getting this configured in VSCode is harder than I anticipated. I also don't want to get into the complexities of setting up Eclipse or IntelliJ for just two files, and Emacs does not support this. The checkstyle CLI is pretty clumsy. I added the braces you suggested and I'll take a bit of time to figure out a better way to handle style.

Reply:
I managed to get checkstyle working, and it found a few other issues that I fixed, such as adding final modifiers where appropriate, using braces in all while/for/if blocks, and removing unnecessary imports. I ignored concerns about line lengths and missing javadocs.